### PR TITLE
Tidy up CHANGELOG in preparation for v2.1 release candidate

### PR DESCRIPTION
**Pull request by MatthewDaggitt**

### CODE REVIEW

**Review by JacquesCarette — APPROVED** [review]

File: agda-stdlib-utils.cabal [hunk 1]
```diff
@@ -1,6 +1,6 @@
 cabal-version:   2.4
 name:            agda-stdlib-utils
-version:         2.0
+version:         2.1
```
Comment (jamesmckinna) [inline agda-stdlib-utils.cabal:3]: Good catch!

File: CHANGELOG.md [hunk 2]
```diff
@@ -496,11 +469,12 @@ Additions to existing modules
   i*j≢0     : .{{_ : NonZero i}} .{{_ : NonZero j}} → NonZero (i * j)
   ```
 
-* In `Data.List.Base` redefine `inits` and `tails` in terms of:
+* In `Data.List.Base` added two new functions:
```
Comment (jamesmckinna) [inline CHANGELOG.md:472]: See #2413 for late-breaking changes!?

**Review by jamesmckinna — APPROVED** [review]
This all looks great, and regularises a lot of ad hoc additions from individual contributors.

Approve, but NB one very recent PR #2413 fixing #2411  might need to be incorporated into your changes after we can agree its final form... (trying to avoid a `CHANGELOG` merge conflict for this PR!?)

File: CHANGELOG.md [hunk 3]
```diff
@@ -789,7 +768,7 @@ Additions to existing modules
   ```
 
 * Added new definition in `Relation.Unary`
-  ```
+  ```agda
   Stable : Pred A ℓ → Set _
   ```
```
Comment (jamesmckinna) [inline CHANGELOG.md:773]: Duplicated in L834 below: suggest deleting L791-L794 outright

**Review by jamesmckinna — COMMENTED** [review]
Additional comments: should go back over the `Additions to existing modules` section to weed out redundant repetitions, and move some other (`Data.X` mostly) blocks into their correct alphabetical positions.

Because they haven't changed in this PR, I couldn't mark them in my review on (web-interface to) GH, nor make suggestions as to changes.

**Comment by jamesmckinna** [comment]
Merging now...

**Pull request merged**